staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -439,3 +441,15 @@ func interpretPatchError(err error) error {
 		return err
 	}
 }
+
+// interpretJSONPatchError interprets the error and returns an error with appropriate HTTP code.
+func interpretJSONPatchError(err error) error {
+	switch {
+	case strings.Contains(err.Error(), "doc is missing"):
+		return errors.NewBadRequest(err.Error())
+	case strings.Contains(err.Error(), "Testing value") && strings.Contains(err.Error(), "failed"):
+		return errors.NewGenericServerResponse(http.StatusUnprocessableEntity, "", schema.GroupResource{}, "", err.Error(), 0, false)
```

Comment:
maybe we could return a more meaningful error here?

Reply:
I would suggest 409 Conflict. From _PATCH Method for HTTP_, 2.2 _Error Handling_ https://tools.ietf.org/html/rfc5789#section-2.2

> Conflicting state: Can be specified with a 409 (Conflict) status
> code when the request cannot be applied given the state of the
> resource. For example, if the client attempted to apply a
> structural modification and the structures assumed to exist did
> not exist (with XML, a patch might specify changing element 'foo'
> to element 'bar' but element 'foo' might not exist).